project/paths.ts

```diff
@@ -7,30 +7,45 @@ import { ParsedArgs } from "../parse_args"
 
 // TODO: Do sourceTsPath and destGdPath have to be relative?
 
-export class Paths {
-  /** Where the .ts files live, e.g. ./src */
+export class Paths implements PathProvider {
   sourceTsPath: string
-
-  /** Where the compiled .gd files go, e.g. ./compiled */
   destGdPath: string
-
-  /** The root path of the project */
   rootPath: string
-
-  /** The full path to the tsconfig file. e.g. /Users/johnfn/GodotProject/tsconfig.json */
   tsconfigPath: string
-
-  /** The path to the Godot definitions folder for unchanging library definitions.
-   * e.g. /Users/johnfn/GodotProject/_godot_defs/static */
   staticGodotDefsPath: string
-
-  /** The path to the Godot definitions folder for definitions based off user files.
-   * e.g. /Users/johnfn/GodotProject/_godot_defs/dynamic */
   dynamicGodotDefsPath: string
-
-  /** The path to the Godot repository, e.g. /Users/johnfn/Godot */
   godotSourceRepoPath: string | undefined
 
+  get csgClassesPath() {
```

Comment:
nit: I tend to prefer functions to getters, just to make things a bit more clear.

Reply:
Will change it.